src/pybind/mgr/ansible/module.py

```diff
@@ -673,6 +694,120 @@ def remove_host(self, host):
+        playbook_operation = PlaybookOperation(client=self.ar_client,
+                                               playbook=SITE_PLAYBOOK,
+                                               logger=self.log,
+                                               result_pattern="",
+                                               params=extravars,
+                                               querystr_dict={"limit": limited})
+
+        # Filter to get the result
+        playbook_operation.output_wizard = ProcessPlaybookResult(self.ar_client,
+                                                                 self.log)
+        playbook_operation.event_filter_list = ["playbook_on_stats"]
+
+        # Execute the playbook
+        self._launch_operation(playbook_operation)
+
+        return playbook_operation
+
+    def remove_stateless_service(self, service_type, id_resource):
+        """ Remove a stateles services providing <sv_id> resources
+
+        :svc_type    : Kind of service (nfs, rgw, mds)
+        :id_resource : Id of the resource provided
+                            <zone name> if service is RGW
+                            ...
+        : returns    : Completion object
+        """
+
+        # Check service_type is supported
+        if service_type not in ["rgw"]:
+            raise orchestrator.OrchestratorError(
+                "{} service not supported".format(service_type))
+
+        # Ansible Inventory group for the kind of service
+        group = "{}s".format(service_type)
+
+        # get the list of hosts where to remove the service
+        # (hosts in resource group)
+        if service_type == "rgw":
+            group_prefix = "rgw_zone_{}"
+
+        resource_group = group_prefix.format(id_resource)
+
+        hosts_list = self.ar_client.get_hosts_in_group(resource_group)
+        limited = ",".join(hosts_list)
+
+        # Avoid manual confirmation
+        extravars = {"ireallymeanit": "yes"}
+
+        # Execute the playbook to remove the service
+        playbook_operation = PlaybookOperation(client=self.ar_client,
+                                               playbook=PURGE_PLAYBOOK,
+                                               logger=self.log,
+                                               result_pattern="",
+                                               params=extravars,
+                                               querystr_dict={"limit": limited})
+
+        # Filter to get the result
+        playbook_operation.output_wizard = ProcessPlaybookResult(self.ar_client,
+                                                                 self.log)
+        playbook_operation.event_filter_list = ["playbook_on_stats"]
+
+        # Cleaning of inventory after a sucessful operation
+        clean_inventory = {}
+        clean_inventory[resource_group] = hosts_list
+        clean_inventory[group] = hosts_list
+        playbook_operation.clean_hosts_on_success = clean_inventory
```

Comment:
Why not simply:
```suggestion
        playbook_operation.clean_hosts_on_success = {'group': hosts_list, 'resource_group': host_lists}
```

Reply:
Because "resource_group" and "group" are variables with the real "group" names where the hosts should be removed.

Reply:
My mistake, but it works also this way:
```suggestion
        playbook_operation.clean_hosts_on_success = {group: hosts_list, resource_group: host_lists}
```